src/index.js

```diff
@@ -278,22 +278,7 @@ module.exports = {
     expect.exportType({
       name: 'DOMNode',
       base: 'object',
-      identify(obj) {
-        return (
-          obj &&
-          obj.nodeName &&
-          [2, 3, 4, 5, 6, 7, 10, 11, 12].indexOf(obj.nodeType) > -1
-        );
-      },
-      equal(a, b) {
-        return a.nodeValue === b.nodeValue;
-      },
-      inspect(element, depth, output) {
-        return output.code(
-          `${element.nodeName} "${element.nodeValue}"`,
-          'prism-string'
-        );
-      }
+      identify: false
```

Comment:
This has some subtle consequences for the node types that we don't have specific types for, such as XML processing instructions.

With this change, the below example fails with `Cannot compare circular structures` because we end up comparing those nodes as objects:

```js
expect(
  parseXml('<?xml-stylesheet type="text/css" href="style.css"?><foo></foo>'),
  'to equal',
  parseXml('<?xml-stylesheet type="text/css" href="style.css"?><foo></foo>')
);
```

That being said, we do a pretty bad job of comparing those as things are now. It seems like the `nodeValue` of processing instruction nodes is always `null` in jsdom :roll_eyes:

The same happens for `<![CDATA[ ... ]]>`. I think we should make sure to support [all the standard node types](https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType) properly before we can get this in.

Reply:
Oh, sorry I thought we were already modeling all of the DOM nodes, let me take a look at that and get back.

Reply:
Let's just make some failing tests, then we can fix those.

Reply:
Hmmmm I also find myself pondering this change a bit. Somehow feels like for semantic purposes you do want to root the different DOM nodes somewhere, but I confess I'm struggling a little with how ti changes the semantics that the "root" of all DOM nodes becomes abstract.

Is the ultimate idea here that we could exclude certain types of DOM nodes from assertions that might not make sense for them (versus having assertions ties to DOMNode)?

Reply:
You can still make assertions on abstract types, so that is not a problem. The real problem right now, is that if you compare a DOMDocument and DOMElement for equality, it will just the equal method on the DOMNode.

Reply:
Ahhhh right, got it! Thanks for explaining this :)